src/main/java/com/epam/aidial/core/controller/DeploymentPostController.java

```diff
@@ -415,7 +421,23 @@ void handleResponse(BufferingReadStream responseStream) {
             tokenUsageFuture = proxy.getTokenStatsTracker().getTokenStats(context).andThen(result -> context.setTokenUsage(result.result()));
         }
 
-        tokenUsageFuture.onComplete(ignore -> {
+        Future<Void> handleResponseFuture;
+        if (context.isStreamingRequest()) {
+            handleResponseFuture = tokenUsageFuture.map(result -> null);
+        } else {
+            handleResponseFuture = tokenUsageFuture.transform(result -> {
```

Comment:
what's the difference between map and transform? why did you chose transform in else block?

Reply:
The handler of `map` is called if `tokenUsageFuture` is completed successfully and transform the result from Future<Token> to Future<Void>.

The handler of `transform` is called unconditionally despite on the future result. We have to process the response any way.

I think we can simplify the code a bit.

Reply:
fixed